main.cpp

```diff
@@ -56,48 +56,77 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* w) : value(v), name(w) {}
+
+    int value;
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+struct Comparison
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b)
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a!=nullptr && b!=nullptr )
```

Comment:
add space around boolean logic operators to make the code easier to read
- [ ] revise